examples/managers/proposal_plugins.py

```diff
@@ -65,39 +69,57 @@ async def main():
     )
 
     payment_manager = PayAllPaymentManager(golem, budget=1.0)
-    demand_manager = AutoDemandManager(
+    demand_manager = RefreshingDemandManager(
         golem,
         payment_manager.get_allocation,
         payload,
     )
-    negotiation_manager = SequentialNegotiationManager(
+    proposal_manager = DefaultProposalManager(
         golem,
         demand_manager.get_initial_proposal,
         plugins=[
-            AddChosenPaymentPlatform(),
-            # class based plugin
-            BlacklistProviderId(BLACKLISTED_PROVIDERS),
-            RejectIfCostsExceeds(1, linear_average_cost),
-            # func plugin
-            blacklist_func,
-            # lambda plugin
-            lambda _, proposal_data: proposal_data.issuer_id not in BLACKLISTED_PROVIDERS,
-            # lambda plugin with reject reason
-            lambda _, proposal_data: RejectProposal(f"Blacklisting {proposal_data.issuer_id}")
-            if proposal_data.issuer_id in BLACKLISTED_PROVIDERS
-            else None,
+            Buffer(
+                min_size=10,
+                max_size=1000,
+                concurrency_size=5,
+            ),
+            BlacklistProviderIdPlugin(BLACKLISTED_PROVIDERS),
+            NegotiatingPlugin(
+                negotiators=[
+                    AddChosenPaymentPlatform(),
+                    # class based plugin
+                    BlacklistProviderIdNegotiator(BLACKLISTED_PROVIDERS),
+                    RejectIfCostsExceeds(1, linear_average_cost),
+                    # func plugin
+                    blacklist_func,  # type: ignore[list-item]
+                    # lambda plugin
+                    lambda _, proposal_data: proposal_data.issuer_id  # type: ignore[list-item]
+                    not in BLACKLISTED_PROVIDERS,
+                    # lambda plugin with reject reason
+                    lambda _, proposal_data: RejectProposal(  # type: ignore[list-item]
+                        f"Blacklisting {proposal_data.issuer_id}"
+                    )
+                    if proposal_data.issuer_id in BLACKLISTED_PROVIDERS
+                    else None,
+                ]
+            ),
+            ScoringBuffer(
+                min_size=3,
+                max_size=5,
+                concurrency_size=3,
+                scorers=[
```

Comment:
What do you mean?

Reply:
Like, the user is required to have context about values range from -1 to 1, without any indication somewhere in, the code. As zen of python - better explicit than implicit - it would be a good idea somewhere to mention that this score is in normalized range.